src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs

```diff
@@ -65,17 +65,10 @@ public async Task TestGetSolutionWithPrimaryFlag(bool fromPrimaryBranch)
             var solutionChecksum = await solution.State.GetChecksumAsync(CancellationToken.None);
             var assetProvider = await GetAssetProviderAsync(workspace, remoteWorkspace, solution);
 
-            var synched = await remoteWorkspace.GetSolutionAsync(assetProvider, solutionChecksum, fromPrimaryBranch, solution.WorkspaceVersion, cancellationToken: CancellationToken.None);
+            var synched = await remoteWorkspace.GetTestAccessor().GetSolutionAsync(assetProvider, solutionChecksum, fromPrimaryBranch, solution.WorkspaceVersion, cancellationToken: CancellationToken.None);
             Assert.Equal(solutionChecksum, await synched.State.GetChecksumAsync(CancellationToken.None));
 
-            if (fromPrimaryBranch)
-            {
-                Assert.IsType<RemoteWorkspace>(synched.Workspace);
-            }
-            else
-            {
-                Assert.IsType<TemporaryWorkspace>(synched.Workspace);
-            }
```

Comment:
this is a change in behavior.  We now never create a TemporaryWorksace on the OOP side.  I believe this is safe and appropraite to do.  The reason we had this in the past was for diagnostics/persistence (with @mavasani  to confirm).  Basically, in the past, the diagnostic system used hte persistence system and would have problems if multiple calls were reading/writing diagnostic data to the persistence store.

However, we changed things so that diagnostics now does not persist anything to disk.  And all roslyn features that *do* persist use checksums now so that there is never an issue if some other call happens to write to the same key+checkum/value data that another service wrote to.